java/sage/MediaServerRemuxer.java

```diff
@@ -0,0 +1,1008 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sage;
+
+import sage.media.format.ContainerFormat;
+import sage.media.format.MPEGParser2;
+import sage.media.format.MediaFormat;
+import sage.media.format.VideoFormat;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MediaServerRemuxer
+{
+  private static final int TS_ALIGN = 188;
+  private static final int MAX_TRANSFER = 33088;
+  private static final int MAX_INIT_BUFFER_SIZE = 20971400;
+  private static final long SWITCH_BYTES_LIMIT = 8388608;
+
+  private static final Map<File, MediaServerRemuxer> remuxerMap =
+      new ConcurrentHashMap<File, MediaServerRemuxer>();
+
+  private boolean closed;
+
+  private final Object switchLock;
+  private boolean switching;
+  private final MediaServer.Connection mediaServer;
+  private String switchFilename;
+  private int switchUploadId;
+  private long switchData;
+  private boolean interAssist;
+
+  private File currentFile;
+  private final RemuxWriter writer;
+  private ByteBuffer writeBuffer;
+  private FileChannel fileChannel;
+  private MPEGParser2.Remuxer2 remuxer2;
+
+  private boolean tsSynced;
+  private int partialTransferIndex;
+  private final byte partialTransfer[];
+
+  private byte[] transferBuffer;
+  private int initOffset;
+  private long bufferIndex;
+  private long bufferLimit;
+  private final AtomicLong fileSize;
+  private ContainerFormat containerFormat;
+  private boolean h264;
+  private boolean mpeg2;
+  private int videoPid;
+
+  private int inputFormat;
+  private int outputFormat;
+  private MPEGParser2.StreamFormat streamFormat;
+  private MPEGParser2.SubFormat subFormat;
+  private MPEGParser2.TuneStringType tuneStringType;
+  private int channel;
+  private int tsid;
+  private int data1;
+  private int data2;
+  private int data3;
+
+  /**
+   * Create a new media server remuxer with auto-detection.
+   * <p/>
+   * The input stream is assumed TS.
+   *
+   * @param fileChannel  The starting FileChannel to be used for writing.
+   * @param outputFormat The format of the remuxed stream.
+   * @param isTV         Is the incoming stream expected to have video?
```

Comment:
This should probably be commented as 'Is the incoming stream from a TV Tuner?' (since at the native level, that's what it is referring to)